### PR TITLE
initial checkin

### PR DESCRIPTION
**Pull request by HarithaIBM**
example usage: https://github.com/reconquest/shdoc

### CODE REVIEW

File: buildenv [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+zopen_append_to_setup()
+{
+  # echo commands that will run when installing via setup.sh
+}
+
+zopen_get_version()
+{
+  # Modify to echo the version of your tool/library
+  # Rather than hardcoding the version, obtain the version by running the tool/library
+  echo "1.0.0"
```
Comment (MikeFultonDev) [inline buildenv]: Can you code the proper version here?

**Review by MikeFultonDev — COMMENTED** [review]
Can you add a read me please?

File: buildenv [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  mkdir -p "${ZOPEN_INSTALL_DIR}/bin/"
+}
+
+zopen_check_results()
+{
+  dir="$1"
+  pfx="$2"
+  chk="$1/$2_check.log"
+
+  # Echo the following information to gauge build health
```
Comment (MikeFultonDev) [inline buildenv:44]: Can you add code to get actual failures here?

**Review by MikeFultonDev — COMMENTED** [review]
Is shdoc needed for another package or is this a tool we want to provide ?

File: buildenv [hunk 3]
```diff
@@ -1,14 +1,16 @@
+SHDOC_VERSION="1.0.0"
 export ZOPEN_STABLE_URL="git@github.com:reconquest/shdoc.git"
```
Comment (IgorTodorovskiIBM) [inline buildenv]: Can we use https protocol to avoid requiring a key?

Comment (MikeFultonDev) [reply buildenv]: don't we have other stable builds that use git@github?

**Comment by HarithaIBM** [comment]
> Can you add a read me please?

Added.

File: buildenv (hunk 1 above)
Comment (HarithaIBM) [reply buildenv]: Done.

File: buildenv (hunk 2 above)
Comment (HarithaIBM) [reply buildenv:44]: I have added a task - https://github.com/ZOSOpenTools/shdocport/issues/3 , Needs gawk issue fix - https://github.com/ZOSOpenTools/gawkport/issues/22

File: buildenv (hunk 3 above)
Comment (HarithaIBM) [reply buildenv]: I have modified to use https, anyways i dont have any hard rule here.

**Comment by HarithaIBM** [comment]
> Is shdoc needed for another package or is this a tool we want to provide ?

This tool used to extract documentation from shell scripts.

**Comment by HarithaIBM** [comment]
Thank you for the review @MikeFultonDev and @IgorTodorovskiIBM . Request you to please review the changes i have added recently.

**Review by MikeFultonDev — APPROVED** [review]
Is this required for another item you are working on? If so I approve. If not I would prefer you for the tests running first before merging.

File: buildenv [hunk 4]
```diff
@@ -0,0 +1,47 @@
+SHDOC_VERSION="1.0.0"
+export ZOPEN_STABLE_URL="https://github.com/ZOSOpenTools/shdocport.git"
```
Comment (IgorTodorovskiIBM) [inline buildenv]: This is pointing to itself, is there an official shdoc repo?

Comment (IgorTodorovskiIBM) [reply buildenv]: Should probably be https://github.com/reconquest/shdoc.git.

Can you add v1.2 as a ZOPEN_STABLE_TAG

**Comment by HarithaIBM** [comment]
> Is this required for another item you are working on? If so I approve. If not I would prefer you for the tests running first before merging.

Sure Mike. This is not a dependent tool. I will work on gawk issue for the tests to proceed here. Thank you.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**